routes/api/categories.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+const express = require("express");
+const HttpStatus = require("http-status-codes");
+
+const interceptors = require("../interceptors");
+const models = require("../../models");
+
+const router = express.Router();
+
+router.get("/", async function (req, res) {
+  const row = await models.Category.findAll();
+  res.json(row);
+});
+
+router.post("/", interceptors.requireLogin, async function (req, res) {
+  const user = req.user;
+  try {
+    let category = await user.getCategory();
```

Comment:
Copying this code from the api/budgets is incorrect. The reason is that User/Budget has a ONE-TO-ONE relationship, so you never can create more than one Budget, so the code checks if that Budget exists already.

However, User/Category is ONE-TO-MANY, so you need to allow for the creation of as many as the User wants/needs.

Also, the User/Category association has not been set up yet in the models, so this code fails.

Reply:
Remove the check for an existing category, and just create a brand new Category as with earlier api route examples...